src/main/java/io/dexi/config/Config.java

```diff
@@ -0,0 +1,187 @@
+        String scheme = uri.getScheme();
+        if ("http".equalsIgnoreCase(scheme)) {
+            builder = builder.configure(properties.setURL(uri.toURL()));
+            configuration = builder.getConfiguration();
+        } else {
+            URL localFileURL = Config.class.getResource(fileLocation);
+            if (localFileURL != null) {
+                builder = builder.configure(properties.setFileName(localFileURL.getFile()));
+                configuration = builder.getConfiguration();
+            }
+        }
+
+        return configuration;
+    }
+
+    private static void addConfigurationToProperties(Configuration configuration) {
+        if (configuration != null) {
+            Iterator<String> keys = configuration.getKeys();
+            while (keys.hasNext()) {
+                String keyWithSection = keys.next();
+
+                String existingValue = properties.getProperty(keyWithSection);
+                if (existingValue == null) {
+                    String value = configuration.getString(keyWithSection);
+                    properties.put(keyWithSection, value);
+                }
+            }
+        }
+    }
+
+    private static void getConfigurationFromURL() throws MalformedURLException, ConfigurationException, URISyntaxException {
+        String dexiAppConfigUrl = System.getenv(DEXI_APP_CONFIG_URL);
+        if (StringUtils.isEmpty(dexiAppConfigUrl)) {
+            dexiAppConfigUrl = System.getProperty(DEXI_APP_CONFIG_URL);
+        }
+
+        if (StringUtils.isNotEmpty(dexiAppConfigUrl)) {
+            Configuration ymlConfigurationURL = getConfigurationFile(dexiAppConfigUrl, YAMLConfiguration.class);
+            addConfigurationToProperties(ymlConfigurationURL);
+        }
+    }
+
+    private static void readLocalConfiguration() throws MalformedURLException, ConfigurationException, URISyntaxException {
+        String fileExtension = localConfigFile.substring(localConfigFile.lastIndexOf(".") + 1);
+
+        Class<? extends FileBasedConfiguration> configurationClass;
+        switch (fileExtension) {
+            case "yml":
+                configurationClass = YAMLConfiguration.class;
+                break;
+            case "json":
+                configurationClass = YAMLConfiguration.class;
+                break;
+            case "xml":
+                configurationClass = YAMLConfiguration.class;
+                break;
+            case "ini":
+                configurationClass = YAMLConfiguration.class;
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported file extension " + fileExtension);
+        }
+
+        Configuration ymlConfigurationLocal = getConfigurationFile(localConfigFile, configurationClass);
+        addConfigurationToProperties(ymlConfigurationLocal);
+    }
+
+    public static void load() throws ConfigurationException, URISyntaxException, MalformedURLException {
+        readLocalConfiguration();
```

Comment:
1. Determine if DEXI_APP_CREDENTIALS is set
1a. Does DEXI_APP_CREDENTIALS contain a file path or a http url (Starts with https?://)
1b. Read from the appropriate location DEXI_APP_CREDENTIALS. If reading from file / url fails throw exception
2. If DEXI_APP_CREDENTIALS not set check for default local file - if not exists throw exception
3. Read env

Reply:
Let's agree on the comment above first, ie. should this configuration utility only be used to read credentials or as a general utility to read configuration files.